packages/protoc-gen-ng/src/output/types/fields/number-message-field.ts

```diff
@@ -104,17 +97,6 @@ export class NumberMessageField implements MessageField {
     }
   }
 
-  printGetter(printer: Printer) {
```

Comment:
If I get it right, these methods go to the abstract class now. The message is however unique, it is always optional; it seems this is lost here. This is also relevant for all other methods of this class.

Please correct me if I'm wrong

Reply:
Sorry, the comment is attached to the wrong place. This goes about message-message-field.ts

https://github.com/ngx-grpc/ngx-grpc/pull/97/files#diff-83e7f03c1f7e78d2a48164ea3200ab966b80d7cac37e08b354412d98b4b8852b

Reply:
Why is this message unique? If we specify the message parameter is not optional? Should not be optional in the interface

Reply:
It is method get. this method is getter. 
Output:
`// Before
 get message(): string | undefined {
    return this._message;
  }
`

`// After
 get message(): string {
    return this._message;
  }
`

Reply:
Please check the specs https://developers.google.com/protocol-buffers/docs/proto3#default

> For message fields, the field is not set. Its exact value is language-dependent. See the [generated code guide](https://developers.google.com/protocol-buffers/docs/reference/overview) for details.

The problem is, if the message is not optional, you would never be able to transfer the tree structures because you fall into an endless recursion of self-invoking message

```proto
message A {
  A a = 1;
}
```

This simple message would not be possible to construct, because at the leaf entry you'd need to pass here no value.

Reply:
You're right! I'll fix this. Are there any further comments?

Reply:
Nope, so far looks good. Thanks!

Reply:
> Nope, so far looks good. Thanks!

I did this.